tests/counter.rs

```diff
@@ -0,0 +1,133 @@
+//! A simple test case using a "counter" type.
+
+use event_listener::Event;
+use futures_lite::future::{block_on, poll_once};
+
+use std::sync::atomic::{fence, AtomicUsize, Ordering};
+use std::sync::{mpsc, Arc, Barrier};
+use std::thread;
+
+struct Counter {
+    counter: AtomicUsize,
+
+    /// Signalled once `counter` has been changed.
+    changed: Event,
+}
+
+impl Counter {
+    fn new() -> Self {
+        Self {
+            counter: AtomicUsize::new(0),
+            changed: Event::new(),
+        }
+    }
+
+    /// Wait for the counter to be incremented.
+    async fn change(&self) -> usize {
+        let original = self.counter.load(Ordering::Acquire);
+        let mut current = original;
+
+        loop {
+            if current != original {
```

Comment:
maybe move this below the "Try again." part

Reply:
The event listener pattern generally assumes that the condition has been already checked before the listener is inserted.